src/RefreshIndicator/RefreshableContainer.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import withStyles from '../styles/withStyles';
+import RefreshIndicator from './RefreshIndicator';
+
+const INDICATOR_SIZE = 40
+const INDICATOR_ACTIVE_POSITION = 114
+
+const styles = {
+  root: {
+    overflowY: 'auto',
+    position: 'relative'
+  },
+  indicator: {
+    position: 'relative',
+    margin: `0 auto -${INDICATOR_SIZE}px`,
+    '&$window': {
+      position: 'fixed'
+    }
+  },
+  window: {}
+}
+
+
+class RefreshableContainer extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = { swiping: false }
+  }
+
+  handleTouchStart = (e) => {
+    if (this.rootNode.scrollTop === 0) {
+      this.setState({ touchStart: e.touches[0].clientY, touchNow: e.touches[0].clientY })
+    }
+  }
+
+  handleTouchMove = (e) => {
+    if (this.rootNode.scrollTop === 0) {
+      const touchNow = e.touches[0].clientY
+      this.setState({ touchNow })
+
+      if (touchNow > this.state.touchStart) {
+        // swiped down
+        e.preventDefault()
```

Comment:
Why do we need event.preventDefault only in case of swipe down?

Reply:
To prevent the browser's swipe-to-refresh function.